nipype/utils/misc.py

```diff
@@ -296,12 +296,16 @@ def dict_diff(dold, dnew, indent=0):
 
     typical use -- log difference for hashed_inputs
     """
-    # First check inputs, since they usually are lists of tuples
-    # and dicts are required.
-    if isinstance(dnew, list):
+    try:
         dnew = dict(dnew)
-    if isinstance(dold, list):
         dold = dict(dold)
```

Comment:
If `dict(dold)` fails while `dict(dnew)` succeeds, then we will get a less comparable error message. What about:

```Python
dnew, dold = dict(dnew), dict(dold)
```

This way, if either fails, the creation of the RHS tuple fails and assignment to the LHS never occurs.